src/Features/Core/Portable/EmbeddedLanguages/Classification/AbstractFallbackEmbeddedLanguageClassifier.cs

```diff
@@ -45,8 +50,15 @@ public void RegisterClassifications(EmbeddedLanguageClassificationContext contex
 
             foreach (var vc in virtualChars)
             {
+                // non-BMP UTF-16 characters' length can be two
                 if (vc.Span.Length > 1)
```

Comment:
❗ The check here should be:

```suggestion
                if (vc.Span.Length > vc.Rune.Utf16SequenceLength)
```

No text operations are necessary.